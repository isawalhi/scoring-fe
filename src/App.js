import React from 'react';
import { Provider } from 'react-redux';
import { Route, BrowserRouter, Switch } from 'react-router-dom';
import createSagaMiddleware from 'redux-saga';
import { createStore, applyMiddleware, compose, combineReducers } from 'redux';
import Layout from './layout/container';
import Sagas from './sagas';
import GameTable from './main/container';
import GameForm from './main/GameForm/container';
import Player from './main/Player/container';
import scoringReducer from './main/reducer';
import LayoutReducer from './layout/reducer';
import { REDUCERS_NAMES } from './constants';
import getRoutes from './routes';
import 'antd/dist/antd.css';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const rootReducer = combineReducers({
  [REDUCERS_NAMES.SCORING_REDUCER]: scoringReducer,
  [REDUCERS_NAMES.LAYOUT_REDUCER]: LayoutReducer,
});

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware))
);

sagaMiddleware.run(Sagas);

const App = () => (
  <Provider store={store}>
    <BrowserRouter>
      <Layout>
        <Switch>
          <Route path={getRoutes.gamesRoute.path} component={GameTable} />
          <Route path={getRoutes.playerRoute.path} component={Player} />
          <Route path={getRoutes.addGameRoute.path} component={GameForm} />
          <Route path="/" exact component={GameTable} />
        </Switch>
      </Layout>
    </BrowserRouter>
  </Provider>
);

export default App;
