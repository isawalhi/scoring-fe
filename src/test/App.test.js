import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../App';

test('renders learn react link', () => {
  const wrapper = shallow(<App />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
