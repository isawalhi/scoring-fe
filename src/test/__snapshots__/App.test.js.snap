// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <Connect(Layout)>
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(GameTable)",
              "type": [Function],
            }
          }
          path="/all-time"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Player)",
              "type": [Function],
            }
          }
          path="/player/:id"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(GameForm)",
              "type": [Function],
            }
          }
          path="/add-game"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(GameTable)",
              "type": [Function],
            }
          }
          exact={true}
          path="/"
        />
      </Switch>
    </Connect(Layout)>
  </BrowserRouter>
</Provider>
`;
